src/transforms/window.d.ts

```diff
@@ -22,19 +22,135 @@ export type WindowReducerFunction = (values: any[]) => any;
 export type WindowReducer = WindowReducerName | WindowReducerFunction;
 
 export interface WindowOptions {
-  k?: number;
+  /**
+   * The size —number of consecutive values— in the window; includes the current
+   * value.
+   */
+  k: number;
+
+  /**
+   * The reducer returns a summary statistic from the **k** values in the
+   * current window, say to compute rolling averages, rolling minimums, or
+   * rolling maximums. The following window reducers are supported:
+   *
+   * * *min* - the minimum
+   * * *max* - the maximum
+   * * *mean* - the mean (average)
+   * * *median* - the median
+   * * *mode* - the mode (most common occurrence)
+   * * *pXX* - the percentile value, where XX is a number in [00,99]
+   * * *sum* - the sum of values
+   * * *deviation* - the standard deviation
+   * * *variance* - the variance per [Welford’s
+   *   algorithm](https://en.wikipedia.org/wiki/Algorithms_for_calculating_variance#Welford's_online_algorithm)
+   * * *difference* - the difference between the last and first window value
+   * * *ratio* - the ratio of the last and first window value
+   * * *first* - the first value
+   * * *last* - the last value
+   * * a function to be passed an array of *k* values
+   *
+   * Defaults to mean.
+   */
   reduce?: WindowReducer;
+
+  /**
+   * How to align the window:
+   *
+   * * *start* - the current value is the first element in the window
+   * * *middle* - the current value is in the middle of the window (rounded to
+   *   the left if the size **k** is even)
+   * * *end* - the current value is the last element in the window.
+   *
+   * Defaults to middle.
+   */
   anchor?: "start" | "middle" | "end";
-  shift?: "leading" | "centered" | "trailing"; // deprecated!
+
+  /** The shift option is deprecated. See **anchor**. */
+  shift?: never; // was "leading" | "centered" | "trailing";
```

Comment:
The **shift** option is deprecated but still supported: existing usage of the option should not cause an error. Changing the type to `never` will cause a `Type 'string' is not assignable to type 'never'` error for TypeScript users:

<img width="497" alt="Screenshot 2023-03-17 at 6 35 27 AM" src="https://user-images.githubusercontent.com/230541/225919975-e7d22e9c-4296-4123-97d4-a775cfeec75c.png">

Instead, we should leave the types as-is, but use the JSDoc `@deprecated` tag. VS Code can then show deprecated features as striked out:

<img width="284" alt="Screenshot 2023-03-17 at 6 41 44 AM" src="https://user-images.githubusercontent.com/230541/225921532-1a43e149-8f22-4555-9025-7fa8da94ab3e.png">

Unfortunately it doesn’t seem to strike out option declarations, though…

Reply:
I understand why we wouldn't want a type error, but the paradoxal consequence is that _shift_ is now actively suggested as a proper autocomplete choice. Let's hope VS Code becomes smarter with this tag?

Reply:
It’s suggested, but striked out. So hopefully it’s clear that it’s supported but deprecated? Seems like this is functioning as desired.

<img width="487" alt="Screenshot 2023-03-17 at 7 54 02 AM" src="https://user-images.githubusercontent.com/230541/225940259-8f4b3063-0c59-431b-8fa7-6db7c36d0ad3.png">

Reply:
All good now (after a restart of VS Code…)